file.c

```diff
@@ -310,5 +474,7 @@ const struct file_operations simplefs_file_ops = {
     .read_iter = generic_file_read_iter,
     .write_iter = generic_file_write_iter,
     .fsync = generic_file_fsync,
-    .open = simplefs_file_open,
+    .open = simplefs_open,
+    .read = simplefs_read,   /*if exist, read_iter will not be called*/
```

Comment:
Does the comment "/*if exist, read_iter will not be called*/" make sense?

Reply:
Is it time to drop `generic_file_read_iter` and `generic_file_write_iter`?

Reply:
I am not sure if it is time to drop these functions.  I need to check more. I traced the read syscall and found the following call sequence: 
`ksys_read` -> `vfs_read`.
```
if (file->f_op->read)
    ret = file->f_op->read(file, buf, count, pos);
else if (file->f_op->read_iter)
    ret = new_sync_read(file, buf, count, pos);
else
    ret = -EINVAL;
```
> vfs_read

Reply:
To test this issue, I used fio and ftrace to check if simplefs and `generic_file_read_iter` are being used. For convenience, I first wrapped `test_generic_file_read_iter`. Here are my test results, which show that `generic_file_read_iter` is indeed not being used.
> wrap generic_file_read_iter
```
static ssize_t test_generic_file_read_iter(struct kiocb *iocb, struct iov_iter *to)
{
    return generic_file_read_iter(iocb, to);
}
```
fio in simplefs filesystem
```
$ fio --name=readtest --size=1M --filename=./testfile --bs=4k --rw=read --iodepth=4
```
ftrace setting
```
echo function > /sys/kernel/debug/tracing/current_tracer
echo test_generic_file_read_iter > /sys/kernel/debug/tracing/set_ftrace_filter
echo simplefs_read > /sys/kernel/debug/tracing/set_ftrace_filter
```

We can see that only simplefs_read is used.
output:
```
root@vm:/sys/kernel/debug/tracing# cat trace
# tracer: function
#
# entries-in-buffer/entries-written: 256/256   #P:4
#
#                                _-----=> irqs-off/BH-disabled
#                               / _----=> need-resched
#                              | / _---=> hardirq/softirq
#                              || / _--=> preempt-depth
#                              ||| / _-=> migrate-disable
#                              |||| /     delay
#           TASK-PID     CPU#  |||||  TIMESTAMP  FUNCTION
#              | |         |   |||||     |         |
             fio-44553   [001] ...1. 83186.031300: simplefs_read <-vfs_read
             fio-44553   [001] ...1. 83186.031325: simplefs_read <-vfs_read
             fio-44553   [001] ...1. 83186.031328: simplefs_read <-vfs_read
             fio-44553   [001] ...1. 83186.031332: simplefs_read <-vfs_read
             ...
             fio-44553   [001] ...1. 83186.031880: simplefs_read <-vfs_read

```
```

Reply:
> We can see that only simplefs_read is used.

It sounds good. Let's move forward.

Reply:
Should I drop them in this PR?

Reply:
> Should I drop them in this PR?

Yes. And you can append the above experiments and observations.